src/vm/amd64/jithelpers_fast.S

```diff
@@ -10,6 +10,45 @@ LEAF_ENTRY JIT_PatchedCodeStart, _TEXT
         ret
 LEAF_END JIT_PatchedCodeStart, _TEXT
 
+
```

Comment:
Due to the relative jump from CheckedWriteBarrier to WriteBarrier, the negative jump overflowed one byte with WriteBarrier increasing in size. Moving it before makes the jump distance independent of both functions' sizes. This order is also consistent with the windows assembly file, making it easier to diff those two.